src/core/utils/urlutils.js

```diff
@@ -82,3 +84,66 @@ export function equivalentParams (params1, params2) {
   }
   return true;
 }
+
+/**
+ * Creates a new params object without params that begin with the given prefixes
+ * @param {SearchParams} params The parameters to remove from
+ * @param {string[]} prefixes The prefixes of parameters to remove
+ * @return {SearchParams} A new instance of SearchParams without removed params
+ *   from the params parameter
+ */
+export function removeParamsWithPrefixes (params, prefixes) {
+  const newParams = new SearchParams();
+  for (const [key, val] of params.entries()) {
+    let includeEntry = true;
+    for (const prefix of prefixes) {
+      if (key.startsWith(prefix)) {
+        includeEntry = false;
+      }
+    }
+    if (includeEntry) {
+      newParams.set(key, val);
+    }
+  }
+  return newParams;
+}
+
+/**
+ * Removes parameters for filters, facets, sort options, and pagination
+ * from the params provided. This is useful for constructing
+ * inter-experience answers links.
+ * @param {SearchParams} params The parameters to remove from
+ * @param {function} getComponentNamesForComponentTypes Given string[]
+ *   component types, returns string[] component names for those types
+ * @return {SearchParams} Parameters that have filtered out params that
+ *   should not persist across the answers experience
+ */
+export function filterParamsForExperienceLink (
+  params,
+  getComponentNamesForComponentTypes
+) {
+  const prefixComponentTypes = [
+    'FACETS',
+    'FILTER_BOX',
+    'FILTER_OPTIONS',
+    'RANGE_FILTER',
+    'DATE_RANGE_FILTER',
+    'SORT_OPTIONS'
+  ].map((component) => {
+    return ComponentTypes[component];
+  });
+  const queryPrefixComponentComponentTypes = [
+    ComponentTypes['GEOLOCATION_FILTER'],
+    ComponentTypes['FILTER_SEARCH']
+  ];
+  let prefixes = getComponentNamesForComponentTypes(prefixComponentTypes);
+  prefixes = prefixes.concat(
+    getComponentNamesForComponentTypes(queryPrefixComponentComponentTypes)
+      .map((name) => { return `${StorageKeys.QUERY}.${name}`; })
+  );
+  prefixes.push(StorageKeys.FILTER);
```

Comment:
Is there anything we can do to simplify this code block a bit and enhance readability? One idea I have is to have:
const filterPrefixes = [
  ComponentTypes.Facets,
   ...,
 `${StorageKeys.QUERY}.${ComponentTypes.Filter_SEARCH}`,
  StorageKeys.FILTER
]
I think that could make this method a bit less complex. It's a bit difficult to follow all the maps, concats, etc. going on and what we're building is a relatively simple list.

Reply:
I don't think that particular prefix map will work because we need to get the names of the component types (ie it would be `${StorageKeys.QUERY}.${getComponentNamesForComponentTypes(['ComponentTypes.Filter_SEARCH'])}` and similarly for all of the other types).

I can remove the map on the prefixes and just spell out all ComponentTypes.<Component> (and also structure the prefixes array a little different). Let me know if we want to simplify further.

I think the root cause is that we're manually recreating the persistentStorage key prefixes instead of sourcing from one place. IE I think these prefixes should be kept somewhere instead of requiring me to construct the component name and then the specific format of this name. However I'm not sure if this is a now thing or a refactor for later on.